core/sdk-core/src/main/java/software/amazon/awssdk/core/traits/XmlNamespaceTrait.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.traits;
+
+import java.util.function.Function;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+
+/**
+ * Trait to include the xml namespace attributes
+ */
+@SdkProtectedApi
+public class XmlNamespaceTrait implements Trait {
```

Comment:
Hmm after thinking about this a bit, can we just make a more generic `XmlAttributeTrait`?

Then when generating the `SdkField` for `Grantee`, we just add two `XmlAttributeTrait`s:

```
.traits(new XmlAttribute("xmlns:xsi", (grantee) -> "http://www.w3.org/2001/XMLSchema-instance")
        new XmlAttribute("xsi:type", (grantee) -> grantee.type()))
```

Reply:
Yeah, like that idea. Let me try it

Reply:
How about this?

```java
    private static final Map<String, AttributeAccessors> ATTRIBUTES_MAP = new LinkedHashMap<>();
    static {
        ATTRIBUTES_MAP.put("xmlns:xsi", AttributeAccessors.builder()
                                                          .attributeGetter((ignore) -> "http://www.w3.org/2001/XMLSchema-instance")
                                                          .build());
        ATTRIBUTES_MAP.put("xsi:type", AttributeAccessors.builder()
                                                         .attributeGetter(t -> ((Grantee) t).typeAsString())
                                                         .build());
    }

    private static final SdkField<Grantee> GRANTEE_FIELD = SdkField
            .<Grantee> builder(MarshallingType.SDK_POJO)
        .getter(getter(Grant::grantee))
        .setter(setter(Builder::grantee))
        .constructor(Grantee::builder)
        .traits(LocationTrait.builder().location(MarshallLocation.PAYLOAD).locationName("Grantee")
                    .unmarshallLocationName("Grantee").build(),
                XmlAttributesTrait.create(ATTRIBUTES_MAP)
                ).build();



```

Reply:
Nice, I like it.